samples/python/console/speech_sample.py

```diff
@@ -92,6 +93,69 @@ def speech_recognize_once_from_file():
             print("Error details: {}".format(cancellation_details.error_details))
     # </SpeechRecognitionWithFile>
 
+def speech_recognize_once_from_file_with_detailed_recognition_results():
+    """performs one-shot speech recognition with input from an audio file, showing detailed recognition results including word-level timing"""
+    # <SpeechRecognitionFromFileWithDetailedRecognitionResults>
+    speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+
+    # Ask for detailed recognition result
+    speech_config.output_format = speechsdk.OutputFormat.Detailed
+
+    # If you also want word-level timing in the detailed recognition results, set the following.
+    # Note that if you set the following, you can omit the previous line
+    #   "speech_config.output_format = speechsdk.OutputFormat.Detailed",
+    # since word-level timing implies detailed recognition results.
+    speech_config.request_word_level_timestamps()
+
+    audio_config = speechsdk.audio.AudioConfig(filename=weatherfilename)
+
+    # Creates a speech recognizer using a file as audio input, also specify the speech language
+    speech_recognizer = speechsdk.SpeechRecognizer(
+        speech_config=speech_config, language="en-US", audio_config=audio_config)
+
+    # Starts speech recognition, and returns after a single utterance is recognized. The end of a
+    # single utterance is determined by listening for silence at the end or until a maximum of 15
+    # seconds of audio is processed. It returns the recognition text as result.
+    # Note: Since recognize_once() returns only a single utterance, it is suitable only for single
+    # shot recognition like command or query.
+    # For long-running multi-utterance recognition, use start_continuous_recognition() instead.
+    result = speech_recognizer.recognize_once()
+
+    # Check the result
+    if result.reason == speechsdk.ResultReason.RecognizedSpeech:
+        print("Recognized: {}".format(result.text))
+
+        # Time units are in hundreds of nanoseconds (HNS), where 10000 HNS equals 1 millisecond
+        print("Offset: {}".format(result.offset))
+        print("Duration: {}".format(result.duration))
+
+        # Now get the detailed recognition results from the JSON
+        json_result = json.loads(result.json)
+
+        # The first cell in the NBest list corresponds to the recognition results 
+        # (NOT the cell with the highest confidence number!)
```

Comment:
Can we put a link to the FAQ in the comments here?

Reply:
@glharper Good idea! I'm going to write a new public doc on detailed reco result, so I will make a separate change in this sample in the Carbon repo to link to the new page, or link to the existing FAQ as you suggested if the new page is taking too long to write.